packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -1618,9 +1618,15 @@ class FileImage extends ImageProvider<FileImage> {
       PaintingBinding.instance.imageCache.evict(key);
       throw StateError('$file is empty and cannot be loaded as an image.');
     }
-    return (file.runtimeType == File)
-        ? decode(await ui.ImmutableBuffer.fromFilePath(file.path))
-        : decode(await ui.ImmutableBuffer.fromUint8List(await file.readAsBytes()));
+    final ui.ImmutableBuffer buffer =
+        (file.runtimeType == File)
+            ? await ui.ImmutableBuffer.fromFilePath(file.path)
+            : await ui.ImmutableBuffer.fromUint8List(await file.readAsBytes());
+    try {
+      return await decode(buffer);
```

Comment:
@jonahwilliams is it always safe to assume that `decode` will not take ownership of the buffer?

(If so, we should update the documentation on ImageProvider)